### ISSUE TITLE
Spacing before subequations

### ISSUE THREAD

**Issue by AlexanderZeilmann** [fmarotta/kaobook#210]
## Summary

There is a [bug in the subequations environment](https://tex.stackexchange.com/questions/27053/too-much-space-between-full-paragraph-and-subequations-env) that causes a too large spacing if the line before fills the entire paragraph.
This can be fixed by adding
```latex
\usepackage{etoolbox}
\preto\subequations{\ifhmode\unskip\fi}
```
to the preamble.

Maybe you want to add this to kaobook.

## MWE
```latex
\documentclass{kaobook}

% \usepackage{etoolbox}
% \preto\subequations{\ifhmode\unskip\fi}

\begin{document}

A line that does not fill the paragraph has normal spacing
\begin{subequations}
	\begin{alignat}{1}
		a \\
		b
	\end{alignat}
\end{subequations}
A line that does fill the paragraph has too much spacing before the eq.
\begin{subequations}
	\begin{alignat}{1}
		a \\
		b
	\end{alignat}
\end{subequations}
A line after the equation.

\end{document}
```
![Bildschirmfoto 2022-02-08 um 17 09 18](https://user-images.githubusercontent.com/901391/153027996-249fe505-0b53-4fd3-afca-a06d11e33f5e.png)

**Comment by fmarotta** [comment]
Done, thanks!

**Comment by MevenBertrand** [comment]
@fmarotta I get an error on my end with your fix, it seems to be because you need to put the fix *after* importing amsmath rather than before. Can you confirm the issue comes from your side?

**Comment by fmarotta** [comment]
Sorry, my bad. I've merged #221, which should fix the problem.
